File: external/packages/bsp/sun60iw2/opt/v4l2_opencv_demo/test_camera.cpp
#include "v4l2_camera.h"
#include <opencv2/opencv.hpp>
#include <chrono>
#include <iostream>

int main(int argc, char* argv[])
{
	const char* device = "/dev/video8";
	if (argc > 1)
		device = argv[1];

	std::string device_str = device;
	V4L2Camera cam(device_str, 640, 480);

	if (!cam.init() || !cam.start()) {
		std::cerr << "Failed to initialize camera" << std::endl;
		return 1;
	}

	cv::namedWindow("Preview", cv::WINDOW_AUTOSIZE);

	int frame_count = 0;
	auto start_time = std::chrono::steady_clock::now();
	double fps = 0.0;

	while (true) {
		cv::Mat frame;
		if (!cam.getFrame(frame))
			break;

		frame_count++;
		auto now = std::chrono::steady_clock::now();
		double elapsed = std::chrono::duration_cast<std::chrono::milliseconds>(now - start_time).count() / 1000.0;
		if (elapsed >= 1.0) {
			fps = frame_count / elapsed;
			start_time = now;
			frame_count = 0;
		}

		char fps_text[32];
		snprintf(fps_text, sizeof(fps_text), "FPS: %.1f", fps);
		cv::putText(frame, fps_text, cv::Point(10, 30),
		            cv::FONT_HERSHEY_SIMPLEX, 0.8, cv::Scalar(0, 255, 0), 2);

		cv::imshow("Preview", frame);
		if (cv::waitKey(1) == 27)
			break;
	}

	return 0;
}

File: external/packages/bsp/sun60iw2/opt/v4l2_opencv_demo/test_camera.py
import cv2
import v4l2cam
import time

cam = v4l2cam.V4L2Camera("/dev/video8", 640, 480)

if not cam.init():
	print("init failed")
if not cam.start():
	print("start failed")

frame_count = 0
start_time = time.time()
fps = 0.0

while True:
	try:
		frame = cam.get_frame()
		frame_count += 1
		now = time.time()
		elapsed = now - start_time

		if elapsed >= 1.0:
			fps = frame_count / elapsed
			frame_count = 0
			start_time = now

		fps_text = f"FPS: {fps:.1f}"
		cv2.putText(frame, fps_text, (10, 30),
					cv2.FONT_HERSHEY_SIMPLEX, 0.8, (0, 255, 0), 2)

		cv2.imshow("Preview", frame)

	except Exception as e:
		print("Error:", e)
		break

	if cv2.waitKey(1) & 0xFF == 27:
		break

cam.stop()
cv2.destroyAllWindows()

